**Parte 1**

1. (10 puntos) Examine la base y elabore una breve descripción de sus variables, distinguiendo
entre dependientes, explicativas, continuas y dummies. ¿Para qué tipo de estudio económico
se emplea habitualmente esta base de datos?

In [6]:
# Cargar librerías
install.packages("wooldridge")
library(wooldridge)
library(dplyr)

set.seed(1)

#Cargar base
data("ceosal1")
ceosal1 <- na.omit(ceosal1)
str(ceosal1)
names(ceosal1)
head(ceosal1)
summary(ceosal1)
?ceosal1

Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)



'data.frame':	209 obs. of  12 variables:
 $ salary  : int  1095 1001 1122 578 1368 1145 1078 1094 1237 833 ...
 $ pcsalary: int  20 32 9 -9 7 5 10 7 16 5 ...
 $ sales   : num  27595 9958 6126 16246 21783 ...
 $ roe     : num  14.1 10.9 23.5 5.9 13.8 ...
 $ pcroe   : num  106.4 -30.6 -16.3 -25.7 -3 ...
 $ ros     : int  191 13 14 -21 56 55 62 44 37 37 ...
 $ indus   : int  1 1 1 1 1 1 1 1 1 1 ...
 $ finance : int  0 0 0 0 0 0 0 0 0 0 ...
 $ consprod: int  0 0 0 0 0 0 0 0 0 0 ...
 $ utility : int  0 0 0 0 0 0 0 0 0 0 ...
 $ lsalary : num  7 6.91 7.02 6.36 7.22 ...
 $ lsales  : num  10.23 9.21 8.72 9.7 9.99 ...
 - attr(*, "time.stamp")= chr "25 Jun 2011 23:03"


[1] "salary"   "pcsalary" "sales"    "roe"      "pcroe"    "ros"     
 [7] "indus"    "finance"  "consprod" "utility"  "lsalary"  "lsales"

,salary,pcsalary,sales,roe,pcroe,ros,indus,finance,consprod,utility,lsalary,lsales
,<int>,<int>,<dbl>,<dbl>,<dbl>,<int>,<int>,<int>,<int>,<int>,<dbl>,<dbl>
1,1095,20,27595.0,14.1,106.4,191,1,0,0,0,6.998509,10.225389
2,1001,32,9958.0,10.9,-30.6,13,1,0,0,0,6.908755,9.206132
3,1122,9,6125.9,23.5,-16.3,14,1,0,0,0,7.022868,8.720281
4,578,-9,16246.0,5.9,-25.7,-21,1,0,0,0,6.359574,9.695602
5,1368,7,21783.2,13.8,-3.0,56,1,0,0,0,7.221105,9.988894
6,1145,5,6021.4,20.0,1.0,55,1,0,0,0,7.043160,8.703075


     salary         pcsalary          sales              roe       
 Min.   :  223   Min.   :-61.00   Min.   :  175.2   Min.   : 0.50  
 1st Qu.:  736   1st Qu.: -1.00   1st Qu.: 2210.3   1st Qu.:12.40  
 Median : 1039   Median :  9.00   Median : 3705.2   Median :15.50  
 Mean   : 1281   Mean   : 13.28   Mean   : 6923.8   Mean   :17.18  
 3rd Qu.: 1407   3rd Qu.: 20.00   3rd Qu.: 7177.0   3rd Qu.:20.00  
 Max.   :14822   Max.   :212.00   Max.   :97649.9   Max.   :56.30  
     pcroe            ros            indus           finance      
 Min.   :-98.9   Min.   :-58.0   Min.   :0.0000   Min.   :0.0000  
 1st Qu.:-21.2   1st Qu.: 21.0   1st Qu.:0.0000   1st Qu.:0.0000  
 Median : -3.0   Median : 52.0   Median :0.0000   Median :0.0000  
 Mean   : 10.8   Mean   : 61.8   Mean   :0.3206   Mean   :0.2201  
 3rd Qu.: 19.5   3rd Qu.: 81.0   3rd Qu.:1.0000   3rd Qu.:0.0000  
 Max.   :977.0   Max.   :418.0   Max.   :1.0000   Max.   :1.0000  
    consprod         utility          lsalary          

Warning message in fetch(key):
“internal error 1 in R_decompress1 with libdeflate”
ERROR while rich displaying an object: Error in fetch(key): lazy-load database '/usr/local/lib/R/site-library/wooldridge/help/wooldridge.rdb' is corrupt

Traceback:
1. sapply(x, f, simplify = simplify)
2. lapply(X = X, FUN = FUN, ...)
3. FUN(X[[i]], ...)
4. tryCatch(withCallingHandlers({
 .     if (!mime %in% names(repr::mime2repr)) 
 .         stop("No repr_* for mimetype ", mime, " in repr::mime2repr")
 .     rpr <- repr::mime2repr[[mime]](obj)
 .     if (is.null(rpr)) 
 .         return(NULL)
 .     prepare_content(is.raw(rpr), rpr)
 . }, error = error_handler), error = outer_handler)
5. tryCatchList(expr, classes, parentenv, handlers)
6. tryCatchOne(expr, names, parentenv, handlers[[1L]])
7. doTryCatch(return(expr), name, parentenv, handler)
8. withCallingHandlers({
 .     if (!mime %in% names(repr::mime2repr)) 
 .         stop("No repr_* for mimetype ", mime, " in repr::mime2repr")
 .     rpr <- rep

**Respuesta**
Dentro de las variables podemos encontrar variables dependientes, variables explicativas continuas y variables explicativas dummy.

Variable dependiente: - Salary
Variables explicativas continuas: - Sales, ROE, ROS
Variables explicativas dummy: - Indus, finance, consprod, utility

Como esta base de datos contiene información de empresa y salarios de cargos, se emplea para estudiar la economía de la empresa, y como se utilizan las variables anteriormente señaladas, esta base de datos se usa para estudiar qué factores explican los salarios.

2. (5 puntos) Genere una variable indicadora (dummy) que tome el valor 1 cuando la variable
sales supere su promedio y 0 en caso contrario. Asigne a esta nueva variable un nombre que
sea fácilmente identificable.


In [12]:
ceosal1$ventas_sobre_promedio <- ifelse(ceosal1$sales > mean(ceosal1$sales), 1, 0)

table(ceosal1$ventas_sobre_promedio)
head(ceosal1[, c("sales", "ventas_sobre_promedio")])


  0   1 
155  54 

,sales,ventas_sobre_promedio
,<dbl>,<dbl>
1,27595.0,1
2,9958.0,1
3,6125.9,0
4,16246.0,1
5,21783.2,1
6,6021.4,0


3. (5 puntos) Cree una variable de la multuplicación entre la variable dummy antes creada y la
variable indus. ¿Qué indica esta interacción?


In [14]:
ceosal1$ventas_sobre_promedio_indus <- ceosal1$ventas_sobre_promedio * ceosal1$indus

table(ceosal1$ventas_sobre_promedio_indus)
head(ceosal1[, c("ventas_sobre_promedio", "indus", "ventas_sobre_promedio_indus")])


  0   1 
188  21 

,ventas_sobre_promedio,indus,ventas_sobre_promedio_indus
,<dbl>,<int>,<dbl>
1,1,1,1
2,1,1,1
3,0,1,0
4,1,1,1
5,1,1,1
6,0,1,0


**Respuesta**
Lo que hace la variable *ventas_sobre_promedio_indus* identifica a las empresas que en conjunto tienen ventas por sobre el promedio y, además, pertenecen al sector industrial. Si cumplen con esa condición, el valor será 1, sino 0.

4. (5 puntos) Cree una variable de la multiplicación entre la variable continua roe y la variable
dummy antes creada. ¿Qué indica esta interacción?


In [15]:
ceosal1$ventas_sobre_promedio_roe <- ceosal1$ventas_sobre_promedio * ceosal1$roe

summary(ceosal1$ventas_sobre_promedio_roe)
head(ceosal1[, c("ventas_sobre_promedio", "roe", "ventas_sobre_promedio_roe")])

   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
  0.000   0.000   0.000   4.251   3.500  43.700 

,ventas_sobre_promedio,roe,ventas_sobre_promedio_roe
,<dbl>,<dbl>,<dbl>
1,1,14.1,14.1
2,1,10.9,10.9
3,0,23.5,0.0
4,1,5.9,5.9
5,1,13.8,13.8
6,0,20.0,0.0


**Respuesta**
Se creó la variable *ventas_sobre_promedio_roe* se creó para separar en grupos y evaluar el ROE en empresas con ventas superiores al promedio y toma valor 0 para las empresas con ventas iguales o inferiores al promedio. El objetivo de esta parte es que el efecto ROE cambie entre empresas con ventas altas y bajas.

5. (5 puntos) Cree una variable que indique si el compañia corresponde a industria (indus) ó si
finance. Asigne a esta nueva variable un nombre que sea fácilmente identificable.

In [16]:
ceosal1$sector_industria_finanzas <- ceosal1$indus + ceosal1$finance

table(ceosal1$sector_industria_finanzas)
head(ceosal1[, c("indus", "finance", "sector_industria_finanzas")])


  0   1 
 96 113 

,indus,finance,sector_industria_finanzas
,<int>,<int>,<int>
1,1,0,1
2,1,0,1
3,1,0,1
4,1,0,1
5,1,0,1
6,1,0,1


**Respuesta**

La variable dummy *sector_industria_finanzas* identifica si la empresa pertenece al sector industrial o financiero. Si es que pertenece a alguno de esos sectores su valor será 1, sino 0.

**Parte 2**

1. (15 puntos) Genere un modelo de regresión lineal, sobre alguna de las variables dependientes,
con respecto a las variables sales, roe y ros. Interprete sus resultados y la significancia de los
regresores. ¿El modelo tiene significancia global al 5%?


In [17]:
# Modelo base
modelo <- lm(salary ~ sales + roe + ros, data = ceosal1)

summary(modelo)


Call:
lm(formula = salary ~ sales + roe + ros, data = ceosal1)

Residuals:
    Min      1Q  Median      3Q     Max 
-1495.0  -486.5  -242.8    98.9 13516.4 

Coefficients:
              Estimate Std. Error t value Pr(>|t|)    
(Intercept) 864.117516 228.399737   3.783 0.000203 ***
sales         0.015482   0.008954   1.729 0.085291 .  
roe          22.003312  11.516554   1.911 0.057454 .  
ros          -1.105191   1.450240  -0.762 0.446892    
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 1360 on 205 degrees of freedom
Multiple R-squared:  0.03191,	Adjusted R-squared:  0.01775 
F-statistic: 2.253 on 3 and 205 DF,  p-value: 0.08337


**Respuesta**

El intercepto estimado fue 864.117516, lo que indica que, cuando sales, roe y ros son 0, el salario esperado es aproximadamente 864.117516 unidades (pesos). Este valor es estadísticamente significativo al 5 % ya que su *p-value* es menor a 0.01 (es de 0.000203).

El coeficiente estimado de sales fue 0.015482, es decir que si se mantiene constante roe y ros, un aumento de una unidad en sales se asocia con un aumento promedio de 0.015482 unidades en salary. Sin embargo, su p-value es 0.085291, por lo que no es estadísticamente significativo al 5 %.

También tenemos el coeficiente estimado de roe que fue de 22, lo que significa que manteniendo constantes sales y ros, un aumento de una unidad en roe se asocia con un aumento promedio de 22 unidades en salary. Su p-value es 0.057454, por lo que no es estadísticamente significativo al 5 %.

Y por último, con respecto a los coeficientes, el coeficiente estimado de ros fue -1.105191, lo que significa que manteniendo constantes sales y roe, un aumento de una unidad en ros se asocia con una disminución promedio de 1.105191 unidades en salary. Su p-value es 0.446892, por lo que no es estadísticamente significativo al 5 %.

Con respecto a su R^2 este tiene un valor de 0.03191 lo que indica que el modelo no es explicativo (valor muy bajo).

2. (10 puntos) Al modelo de regresión lineal anterior, agregue las variables creadas en el inciso 1.2
y 1.4, interprete sus resultados. ¿Cómo afectan los regresores de estas variables en el salario?

In [19]:
modelo <- lm(salary ~ sales + roe + ros + ventas_sobre_promedio + ventas_sobre_promedio_roe,
              data = ceosal1)

summary(modelo)


Call:
lm(formula = salary ~ sales + roe + ros + ventas_sobre_promedio + 
    ventas_sobre_promedio_roe, data = ceosal1)

Residuals:
    Min      1Q  Median      3Q     Max 
 -976.0  -469.1  -263.4    97.2 13566.3 

Coefficients:
                            Estimate Std. Error t value Pr(>|t|)   
(Intercept)               783.298711 258.064395   3.035  0.00272 **
sales                       0.006657   0.011197   0.594  0.55285   
roe                        24.472338  13.088222   1.870  0.06295 . 
ros                        -0.986904   1.457818  -0.677  0.49919   
ventas_sobre_promedio     513.083213 510.134319   1.006  0.31572   
ventas_sobre_promedio_roe  -9.498301  25.587585  -0.371  0.71087   
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 1361 on 203 degrees of freedom
Multiple R-squared:  0.04059,	Adjusted R-squared:  0.01696 
F-statistic: 1.718 on 5 and 203 DF,  p-value: 0.132


**Respuesta**

El intercepto estimado fue 783.298711, lo que significa que cuando sales, roe, ros y las variables asociadas a la dummy toman valor 0, el salario esperado es aproximadamente 783.298711 unidades. El intercepto es estadísticamente significativo al 5 %, ya que su p-value es 0.00272.

El coeficiente de sales fue 0.006657, lo que significa que, manteniendo constantes las demás variables, un aumento de una unidad en sales se asocia con un aumento promedio de 0.006657 unidades en salary. No es estadísticamente significativo al 5 %.

El coeficiente de roe fue 24.472338, lo que indica que, para las empresas con ventas_sobre_promedio = 0, un aumento de una unidad en roe se asocia con un aumento promedio de 24.472338 unidades en salary, si es que se mantienen constantes las demás variables. Su p-value es 0.06295, por lo que no es estadísticamente significativo al 5 %.

El coeficiente de ros fue -0.986904, lo que significa que si se mantienen constantes las demás variables, un aumento de una unidad en ros se asocia con una disminución promedio de 0.987 unidades en salary. Su p-value es 0.49919, por lo que no es significativo al 5 %.

El coeficiente de la dummy ventas_sobre_promedio fue 513.083213, lo que significa que cuando roe = 0, las empresas con ventas superiores al promedio presentan un salario esperado aproximadamente 513.083213 unidades mayor que las empresas con ventas iguales o inferiores al promedio, manteniendo constantes las demás variables. Su p-value es 0.31572, por lo que no es estadísticamente significativo al 5 %.

El coeficiente de la interacción ventas_sobre_promedio_roe fue -9.498301, lo que significa que el efecto de roe sobre salary es 9.498301 unidades menor en las empresas con ventas superiores al promedio, en comparación con las empresas con ventas iguales o inferiores al promedio. Su p-value es 0.71087, por lo que no existe evidencia estadística suficiente al 5% para afirmar que el efecto de roe difiera entre ambos grupos.

El  R^2 también tiene un valor bajo por lo que el modelo no es explicativo.

3. (15 puntos) Realiza un test de hipotesis para evaluar si las dos variables que se agregaron son
significativas para el modelo simultaneamente. Use significancia del 5%.

In [20]:
F_critico <- qf(0.95, 2, 203)

R2_r <- 0.03191
R2_ur <- 0.04059

F_est <- ((R2_ur - R2_r) / (1 - R2_ur)) * (203 / 2)

F_est
F_critico

if (F_est > F_critico) {
  "Se rechaza H0"
} else {
  "No se rechaza H0"
}

[1] 0.9182935

[1] 3.040379

[1] "No se rechaza H0"

**Respuesta**

Se tiene que *F_est* es 0.91829353456812 y *F_critico* es 3.04037936353184, Como *F_est*<*F_critico no se rechaza la hipótesis nula al 5 %. Esto quiere decir que no existe evidencia estadística suficiente para concluir que las variables ventas_sobre_promedio y ventas_sobre_promedio_roe sean conjuntamente significativas en la explicación del salario.

4. (15 puntos) Volviendo a la regresión del salario en función de sales, roe y ros. Considere
dos grupos, Grupo 1: Aquellos clientes que sales es mayor al promedio, y Grupo 2: Aquellos
clientes que sales es menor al promedio. Realice un test de hipotesis para grupos al 5% de
significancia. Use el metodo que más le acomode.


In [21]:
# Crear interacciones para comparar grupos
ceosal1$sales_D <- ceosal1$sales * ceosal1$ventas_sobre_promedio
ceosal1$roe_D   <- ceosal1$roe   * ceosal1$ventas_sobre_promedio
ceosal1$ros_D   <- ceosal1$ros   * ceosal1$ventas_sobre_promedio

modelo3 <- lm(salary ~ sales + roe + ros + ventas_sobre_promedio + sales_D + roe_D + ros_D,
              data = ceosal1)

summary(modelo3)

# Test F
F_critico <- qf(0.95, 4, 201)

R2_ur <- 0.06028
R2_r  <- 0.03191

F_est <- ((R2_ur - R2_r) / (1 - R2_ur)) * (201 / 4)

F_est
F_critico

if (F_est > F_critico) {
  "Se rechaza H0"
} else {
  "No se rechaza H0"
}


Call:
lm(formula = salary ~ sales + roe + ros + ventas_sobre_promedio + 
    sales_D + roe_D + ros_D, data = ceosal1)

Residuals:
    Min      1Q  Median      3Q     Max 
-1024.7  -456.7  -210.2    83.2 13712.4 

Coefficients:
                        Estimate Std. Error t value Pr(>|t|)  
(Intercept)            331.86179  352.76387   0.941   0.3480  
sales                    0.13362    0.06635   2.014   0.0454 *
roe                     25.63426   13.07068   1.961   0.0512 .
ros                     -0.47570    1.66943  -0.285   0.7760  
ventas_sobre_promedio 1043.02106  578.05672   1.804   0.0727 .
sales_D                 -0.13033    0.06731  -1.936   0.0542 .
roe_D                  -17.86857   28.02885  -0.638   0.5245  
ros_D                    1.55160    3.84565   0.403   0.6870  
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 1353 on 201 degrees of freedom
Multiple R-squared:  0.06028,	Adjusted R-squared:  0.02756 
F-statistic: 1.842 on

[1] 1.51704

[1] 2.416574

[1] "No se rechaza H0"

**Respuesta**

Como *F_est*<*F_critico no se rechaza la hipótesis nula al 5 %. Esto quiere decir que no existe evidencia estadística suficiente para afirmar que el modelo salarial difiera entre empresas con ventas altas y empresas con ventas bajas.

5. (15 puntos) Usando la regresión del salario en función de sales, roe y ros, se distinguen los
grupos mediante las dummies; indus, finance, consprod y utility, que corresponden al tipo de
empresa. Realice mediante test de Chow, con 5% de significancia, si los modelos para cada
grupo son iguales.

In [22]:
ceosal1_indus    <- subset(ceosal1, indus == 1)
ceosal1_finance  <- subset(ceosal1, finance == 1)
ceosal1_consprod <- subset(ceosal1, consprod == 1)
ceosal1_utility  <- subset(ceosal1, utility == 1)

modelo_r <- lm(salary ~ sales + roe + ros, data = ceosal1)

modelo_indus    <- lm(salary ~ sales + roe + ros, data = ceosal1_indus)
modelo_finance  <- lm(salary ~ sales + roe + ros, data = ceosal1_finance)
modelo_consprod <- lm(salary ~ sales + roe + ros, data = ceosal1_consprod)
modelo_utility  <- lm(salary ~ sales + roe + ros, data = ceosal1_utility)


SSR_r <- sum(residuals(modelo_r)^2)

SSR_indus    <- sum(residuals(modelo_indus)^2)
SSR_finance  <- sum(residuals(modelo_finance)^2)
SSR_consprod <- sum(residuals(modelo_consprod)^2)
SSR_utility  <- sum(residuals(modelo_utility)^2)

SSR_ur <- SSR_indus + SSR_finance + SSR_consprod + SSR_utility

G  <- 4      # número de grupos
k  <- 3      # número de regresores (sales, roe, ros)
n  <- nrow(ceosal1)

gl1 <- (G - 1) * (k + 1)
gl2 <- n - G * (k + 1)

# F crítico
F_critico <- qf(0.95, gl1, gl2)

# Estadístico F de Chow
F_est <- ((SSR_r - SSR_ur) / gl1) / (SSR_ur / gl2)

F_est
F_critico

if (F_est > F_critico) {
  "Se rechaza H0"
} else {
  "No se rechaza H0"
}

[1] 1.398349

[1] 1.802614

[1] "No se rechaza H0"

Como *F_est*<*F_critico no se rechaza la hipótesis nula al 5 %. Por lo tanto, no existe evidencia estadística suficiente para afirmar que los modelos difieren entre los cuatro sectores.